src/Core/Web/XMLHttpRequestHelper.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generated On: 2015-10-5
+ * Class: XMLHttpRequestHelper. Simple promise-based wrapper around XMLHttpRequest
+ */
+
+
+define('Core/Web/XMLHttpRequestHelper', ['when'], function(when) {
+    return {
+        fetch: function(url, responseType) {
```

Comment:
How about using the native `fetch` when available and a polyfill otherwise? http://caniuse.com/#feat=fetch

(and similarly for promises, now's the time)